sdk/python/feast/type_map.py

```diff
@@ -402,7 +402,10 @@ def _python_value_to_proto_value(
             valid_scalar_types,
         ) = PYTHON_SCALAR_VALUE_TYPE_TO_PROTO_VALUE[feast_value_type]
         if valid_scalar_types:
-            assert type(sample) in valid_scalar_types
+            if sample == 0 or sample == 0.0:
```

Comment:
nit: would you mind adding a comment indicating why this special case is necessary?

Reply:
@kysersozelee hm I think this comment is still not super clear to me - it would be great if you could add _why_ type validation cannot be performed with only one type (if I understand correctly from #3400, the issue is because you might e.g. have an int value in a float column)

Reply:
@felixwang9817 

Left the two comment. Let me know if it is not inappropriate!
```python
# Numpy convert 0 to int. However, in the feature view definition, the type of column may be a float.
# So, if value is 0, type validation must pass if scalar_types are either int or float.
```

Reply:
thanks @kysersozelee!